### PR TITLE
CLDR-16172 support empty string values in the FE

### PR DESCRIPTION
**Pull request by srl295**
- had to do some work arounds for abstention as well, also for JSON serialization

CLDR-16172

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/DataPage.java [hunk 1]
```diff
@@ -858,7 +858,7 @@ public String getWinningVHash() {
         }
 
         public String getVoteVHash() {
-            String voteVhash = "";
+            String voteVhash = null;  // abstension
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/DataPage.java:861]: comment -- spelling should be "abstention"

File: tools/cldr-apps/js/src/esm/cldrTable.js [hunk 2]
```diff
@@ -419,7 +424,7 @@ function reallyUpdateRow(tr, theRow) {
   tr.rawValueToItem = {}; // hash:  string value to item (which has a div)
   for (let k in theRow.items) {
     const item = theRow.items[k];
-    if (item.value) {
+    if (item.value || item.value === "") {
```
Comment (btangmu) [inline tools/cldr-apps/js/src/esm/cldrTable.js:427]: ideally encapsulate this as a boolean method -- maybe `cldrVote.hasValue(item.value)`  -- as it is the logic is duplicated half a dozen times

**Review by btangmu — APPROVED** [review]
I'm concerned these changes may have missed other places where the front or back end, or their http requests/responses, make inconsistent assumptions about null/empty/undefined values... For example, are Abstain and empty-string votes distinguished in the database, and for Import Old Votes?

**Comment by srl295** [comment]
Abstain is already handled as null in the DB. Null vs empty string seems to transit fine.  I think that these changes are fairly contained.

**Pull request merged**